### ISSUE TITLE
feat: Add Save option to VerseMatch results

### ISSUE THREAD

**Issue by JustinhSE** [Namit2111/bible-verse-finder#75]
## NOTE:
- #55 must be done before this issue is started. 

### Overview

Implement a "Save" feature for individual verse outputs in the VerseMatch component. Users should be able to save specific verses from the search results for later reference. This feature will be accessible through a "More" button next to each verse output, which will include a "Save" option. When a user saves a verse, it will be stored in their user profile under a "Saved" section in a list format.

### Tasks

- [ ] Update the VerseMatch component to include a "More" button next to each verse output
  - Add a "Save" option to the menu
  
- [ ] Implement the save functionality
  - Create a new Firebase function to handle saving verses to the user's profile
  - Update the user's document in Firestore to include a "saved" array

- [ ] Add error handling and loading states for the save operation
  - Display a loading indicator while the save operation is in progress
  - Show success/error messages to the user after the save attempt

**Comment by Agastya18** [comment]
assign it to me @JustinhSE

**Comment by JustinhSE** [comment]
@Agastya18 as it says in the issue, this is currently on hold
